src/realm/object-store/c_api/types.hpp

```diff
@@ -501,6 +501,21 @@ struct realm_refresh_callback_token : realm_callback_token {
     ~realm_refresh_callback_token() override;
 };
 
+struct realm_async_transaction_token : realm::c_api::WrapC {
```

Comment:
I don't thing making a whole new heap-allocated type just for the async handle integer is worth it. It's not something we need to keep around like notification tokens, it doesn't have any logic in and of itself except holding an unsigned int value, and its lifetime is pretty short, if existing altogether (since it's fine for SDKs to ignore the async handle return if they don't intend to cancel async operations).